python/packages/ai/teams/ai/planner/openai_planner.py

```diff
@@ -144,25 +153,71 @@ async def _complete_prompt(
 
         result: str
 
-        # TODO: implement chat completion
-        result = await self._complete_text(prompt_template)
+        if is_chat_completion:
+            user_message = turn_state.temp.value.input
+            result = await self._complete_chat(
+                prompt_template, turn_state, user_message, history_options
+            )
+        else:
+            result = await self._complete_text(prompt_template)
 
         duration = datetime.now() - start_time
         # TODO: investigate how to get prompt/completion tokens
         self._log_request(f"\n{log_prefix} SUCCEEDED: duration={duration} response={result}")
 
         return result
 
+    async def _complete_chat(
+        self,
+        prompt_template: PromptTemplate,
+        state: TurnState,
+        user_message: str,
+        options: AIHistoryOptions,
+    ):
+        prompt_template_config = generate_sk_prompt_template_config(prompt_template)
+        request_settings = ChatRequestSettings.from_completion_config(
+            prompt_template_config.completion
+        )
+
+        chat_completion_client = self._sk.get_ai_service(ChatCompletionClientBase)(self._sk)
+
+        chat_history: List[Tuple[str, str]] = []
```

Comment:
chat history is now in another class so please use that one, it exposes many helper methods that should make this work easier.

Reply:
created task for this https://github.com/microsoft/teams-ai/issues/444

Reply:
Since the code is both on active branch. If you don't mind and feels fine with my other comments, could merge my PR first and have additional commit to your PR to leverage ChatHistory class. I can help do this if you need.